hs_core/hydroshare/hs_bagit.py

```diff
@@ -29,55 +27,47 @@ def delete_bag(resource):
     :param resource: the resource to delete the bag for.
     :return: none
     """
-    res_id = resource.short_id
-
-    res_path = resource.root_path
-    bagname = 'bags/{res_id}.zip'.format(res_id=res_id)
     istorage = resource.get_irods_storage()
-    if resource.resource_federation_path:
-        bagname = '{}/{}'.format(resource.resource_federation_path, bagname)
 
     # delete resource directory first to remove all generated bag-related files for the resource
-    if istorage.exists(res_path):
-        istorage.delete(res_path)
+    # TODO: Is this correct? Is it really necessary? It seems this deletes all content!
+    if istorage.exists(resource.root_path):
+        istorage.delete(resource.root_path)
 
-    # the resource bag may not exist due to on-demand bagging
-    if istorage.exists(bagname):
-        # delete the resource bag
-        istorage.delete(bagname)
+    if istorage.exists(resource.bag_path):
+        istorage.delete(resource.bag_path)
 
     # delete the bags table
+    # TODO: Why were extra bags created in the first place?
     for bag in resource.bags.all():
         bag.delete()
```

Comment:
@alvacouch This Bags objects on resources don't hold anything meaningful at this point. It exists as part of HydroShare legacy before we moved bag creation to on-demand bag creation on iRODS side. It can be removed in my opinion. I remembered I intended to remove it, but there were some issues probably migration-related, so I did not really take it on at the time.